### PR TITLE
Update CI to Xcode 9

### PR DESCRIPTION
**Pull request by endocrimes**
- [x] Upgrades circle.yml to 9.0.x

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: fastlane_core/lib/fastlane_core/device_manager.rb [hunk 1]
```diff
@@ -251,11 +251,15 @@ def copy_logfile(device, log_identity, logs_destination_dir)
...
         logarchive_dst = Shellwords.escape(File.join(logs_destination_dir, "system_logs-#{log_identity}.logarchive"))
         FileUtils.rm_rf(logarchive_dst)
-        command = "xcrun simctl spawn #{device.udid} log collect --output #{logarchive_dst} 2>/dev/null"
-        FastlaneCore::CommandExecutor.execute(command: command, print_all: false, print_command: true)
+  #      FileUtils.mkdir_p(logs_destination_dir)
+        command = "xcrun simctl spawn #{device.udid} log collect --output #{logarchive_dst}"
+        begin
+          FastlaneCore::CommandExecutor.execute(command: command, print_all: true, print_command: true)
+        rescue FastlaneCore::Interface::FastlaneError
+          UI.error "Failed to copy simulator logs"
```
Comment (lyndsey-ferguson) [inline fastlane_core/lib/fastlane_core/device_manager.rb]: I am using fastlane with Xcode 8.3.2 and the only problem I experienced was that if the logarchive already exists, this command fails: which makes sense to me. What are you seeing?

Comment (DanToml) [inline fastlane_core/lib/fastlane_core/device_manager.rb]: i'm seeing an issue whereby in some cases simctl fails to copy a nonexistent diagnostics dir, but still copies the rest of the archive.

maybe @russbishop knows more?

**Comment by ohayon** [comment]
Hey everyone, just curious what we can do to push this forward, it would be great to get _fastlane_ building with up-to-date machines! 🚀

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Gym Gym::DetectValues Xcode config handling fetches the custom build path from the Xcode config | <a href='https://github.com/fastlane/fastlane/blob/cf6077b041e7bb15ecc644ffbc8bd1f55dfc7ce5/./gym/spec/detect_values_spec.rb'>detect_values_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by DanToml** [comment]
@KrauseFx Green!

**Review by KrauseFx — APPROVED** [review]
Beautiful ❤️

**Comment by KrauseFx** [comment]
🚀

**Pull request merged**